tests/backwards_compatibility_tests/compatibility_test_runner.py

```diff
@@ -0,0 +1,603 @@
+import argparse
+import importlib
+import pkgutil
+import time
+
+import pytest
+from typing import Optional, Set
+import subprocess
+import sys
+import os
+import requests
+import semver
+import traceback
+
+from compatibility_test_logger import get_logger
+
+marqo_transfer_state_version = semver.VersionInfo.parse("2.9.0")
+
+sys.path.append(os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__)))))
+from base_test_case import BaseCompatibilityTestCase
+
+# Keep track of containers that need cleanup
+containers_to_cleanup: Set[str] = set()
+volumes_to_cleanup: Set[str] = set()
+
+logger = get_logger(__name__)
+
+def load_all_subclasses(package_name):
+    logger.debug(f"Inside load_all_subclasses with package_name: {package_name}")
+    package = importlib.import_module(package_name)
+    package_path = package.__path__
+
+    for _, module_name, _ in pkgutil.iter_modules(package_path):
+        importlib.import_module(f"{package_name}.{module_name}")
+
+
+def pull_remote_image_from_ecr(image_digest: str):
```

Comment:
Not necessary to change in this PR, but maybe consider using docker python SDK docker-py (https://docker-py.readthedocs.io/en/stable/) to replace the subprocess call?

Reply:
Refactoring suggestion: consider extract all docker related logic to its own class or python file for separation of concerns.

Reply:
Will try incorporating in the next PR